File: twoSum.js
class TwoSum{
    constructor(){
        this.dict={}
    }
    add(number){
        if (number in this.dict){
            this.dict[number]+=1
        }
        else{
            this.dict[number]=1
        }
    }
    find(value){
        let r=this.dict
        for (let num in r){
            let left=value-num
            if ((left!==num && left in r ) || (left===num && r[num]>1)){
                return true
            }
            
        }
        return false
    }
}

obj=new TwoSum()
obj.add(4)
obj.add(5)
obj.add(4)
console.log(obj.find(7))

File: twoSum.py
# import collections
class TwoSum(object):
    def __init__(self):
        # self.record = collections.defaultdict(int)
        self.record={}
    def add(self, number):
        if number in self.record:
            self.record[number]+=1
        else:
            self.record[number]=1
    def find(self, value):
        r=self.record
        for num in r:
            left = value - num
            if left!=num and left in r or left == num and r[num]>1:
                return True
        return False
    
obj=TwoSum()
obj.add(4)
obj.add(5)
obj.add(4)
print(obj.find(8))